src/utils/Firebase.tsx

```diff
@@ -241,16 +243,72 @@ export async function createNewProject(
 
     const docRef = await addDoc(collection(db, 'Projects'), newProject)
 
-    // console.log('Document written with ID: ', docRef.id) // Debug log
+    console.log('Document written with ID: ', docRef.id) // Debug log
     return docRef.id
   } catch (error) {
-      if (error instanceof Error) {
-        const errorMessage = "Failed to create Project. Status: ${error.name}. Message: ${error.message}";
-        toast.error(errorMessage);
-        throw new Error(errorMessage);
+    if (error instanceof Error) {
+      const errorMessage =
+        'Failed to create Project. Status: ${error.name}. Message: ${error.message}'
+      console.error(errorMessage)
+      throw new Error(errorMessage)
     } else {
-        const genericErrorMessage = "An unexpected error occurred while creating the project";
-        toast.error(genericErrorMessage);
+      const genericErrorMessage =
+        'An unexpected error occurred while creating the project'
+      console.error(genericErrorMessage)
+      throw new Error(genericErrorMessage)
     }
   }
-}
\ No newline at end of file
+}
+
+// Edit a Portfolio Project
+
+export async function editProject(
+  projectID: string,
+  userId: string,
+  title: string,
+  thumbnail: string,
+  description: string,
+  url: string
+) {
+  if (!projectID || !userId || !title) {
+    console.error(
+      'Project ID, User ID, and title are required to edit a project'
+    )
+    return
+  }
+  try {
+    const projectDocRef = doc(db, 'Projects', projectID)
+
+    const updatedFields = {
+      userId: userId,
+      title: title,
+      thumbnail: thumbnail,
+      description: description,
+      url: url
+    }
+
+    await updateDoc(projectDocRef, updatedFields)
+    console.log('Project updated successfully')
+    return updatedFields
+  } catch (error) {
+    console.error('Error updating project:', error)
+    throw error
+  }
+}
+
+// Get all projects by a specific user
+export async function getAllProjectsByUserID(userID: string) {
+  try {
+    const projectsRef = collection(db, 'Projects')
+    const q = query(projectsRef, where('userID', '==', userID))
+    const querySnapshot = await getDocs(q)
+    const projects = querySnapshot.docs.map(doc => ({
+      id: doc.id,
```

Comment:
maybe you can tack the id on at the end to be cleaner

Reply:
✅